### PR TITLE
Node - Week6 - Assignment

### PR DESCRIPTION
**Pull request by AnnaSolovykh**

### CODE REVIEW

File: 03-task-manager/starter/controllers/tasks.js
Comment (leo-mutuku) [inline 03-task-manager/starter/controllers/tasks.js:1]: Try catch error control added correctly.

File: 03-task-manager/starter/models/Task.js
Comment (leo-mutuku) [inline 03-task-manager/starter/models/Task.js:1]: Schema field validation added correctly

File: 03-task-manager/starter/controllers/tasks.js
Comment (leo-mutuku) [inline 03-task-manager/starter/controllers/tasks.js:1]: commend well added

File: 03-task-manager/starter/app.js
Comment (leo-mutuku) [inline 03-task-manager/starter/app.js:1]: Middleware well implimented.

# NOTE 1
The not found middleware should be place last after all the other routes, else the other routes will not be reachable as the latter excutes to true.

File: 03-task-manager/starter/middleware/not-found.js
Comment (leo-mutuku) [inline 03-task-manager/starter/middleware/not-found.js:1]: Not found routes,404 error well implimented

File: 03-task-manager/starter/middleware/async.js
Comment (leo-mutuku) [inline 03-task-manager/starter/middleware/async.js:1]: A wrapper on async await well implimented.

File: 03-task-manager/starter/controllers/tasks.js
Comment (leo-mutuku) [inline 03-task-manager/starter/controllers/tasks.js:1]: Async await wrapper well implimented on api routes

File: 03-task-manager/starter/middleware/error-handler.js
Comment (leo-mutuku) [inline 03-task-manager/starter/middleware/error-handler.js:1]: Error handling middleware well implimented

File: 03-task-manager/starter/errors/custom-error.js
Comment (leo-mutuku) [inline 03-task-manager/starter/errors/custom-error.js:1]: custum error class well implimented

File: 03-task-manager/starter/app.js
Comment (leo-mutuku) [inline 03-task-manager/starter/app.js:1]: Env variable with a fail inclusive port well added

**Review by leo-mutuku — APPROVED** [review]

**Comment by AnnaSolovykh** [comment]
@leo-mutuku thank you for the review!

**Pull request merged**